lib/top-bar.jsx

```diff
@@ -0,0 +1,94 @@
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux'
+
+import { setRevisionState } from './state/revisions/actions';
+import { toggleLayoutType } from './state/settings/actions'
+import NewNoteIcon from './icons/new-note';
+import NoteToolbar from './note-toolbar';
+
+class TopBar extends Component {
+
+	static propTypes = {
+		layoutType: PropTypes.string,
+		revisionState: PropTypes.object,
+		onToggle: PropTypes.func,
+		onRevisionChange: PropTypes.func,
+		note: PropTypes.object,
+		onNewNote: PropTypes.func,
+		onTrashNote: PropTypes.func,
+		onRestoreNote: PropTypes.func,
+		onShareNote: PropTypes.func,
+		onDeleteNoteForever: PropTypes.func,
+		onRevisions: PropTypes.func,
+		onCloseNote: PropTypes.func,
+		onNoteInfo: PropTypes.func,
+	}
+
+	toggleIcon() {
+		switch ( this.props.layoutType ) {
+			case '3-col' : return '3 COLUMNS';
+			case '2-col' : return '2 COLUMNS';
+			case '1-col' : return 'EDITOR';
+		}
```

Comment:
this is another type of data to ponder. what is the essence of what we are trying to represent here?

is it really one, two, or three columns?

might it be the presence or absence of different UI components at a particular time?

while I'm not saying this is the only way to do it, at first thought I think about having a set wherein membership in that set indicates that the component should be visible. it also ties the state to the components themselves in an extendible way instead of hard-coding the meaning of columns to components.

check out `lodash.union()` and `lodash.difference()` too. we have native-JS sets now but those are unfortunately mutable objects 😢 

```js
const visiblePanels = ( state = [], action ) => get( {
	HIDE_PANEL: ( { panelName } ) => difference( state, [ panelName ] ),
	SHOW_PANEL: ( { panelName } ) => union( state, [ panelName ] ),
}, action.type, () => state )( action );
```